pkg/api/persistentvolume/util.go

```diff
@@ -31,16 +31,28 @@ const (
 	deprecatedStorageClassAnnotationsMsg = `deprecated since v1.8; use "storageClassName" attribute instead`
 )
 
-// DropDisabledFields removes disabled fields from the pv spec.
+// DropDisabledSpecFields removes disabled fields from the pv spec.
 // This should be called from PrepareForCreate/PrepareForUpdate for all resources containing a pv spec.
-func DropDisabledFields(pvSpec *api.PersistentVolumeSpec, oldPVSpec *api.PersistentVolumeSpec) {
+func DropDisabledSpecFields(pvSpec *api.PersistentVolumeSpec, oldPVSpec *api.PersistentVolumeSpec) {
 	if !utilfeature.DefaultFeatureGate.Enabled(features.CSINodeExpandSecret) && !hasNodeExpansionSecrets(oldPVSpec) {
 		if pvSpec.CSI != nil {
 			pvSpec.CSI.NodeExpandSecretRef = nil
 		}
 	}
 }
 
+// DropDisabledStatusFields removes disabled fields from the pv status.
+// This should be called from PrepareForUpdate for all resources containing a pv status.
+func DropDisabledStatusFields(oldStatus, newStatus *api.PersistentVolumeStatus) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.PersistentVolumeLastPhaseTransitionTime) && oldStatus.LastPhaseTransitionTime.IsZero() {
+		newStatus.LastPhaseTransitionTime = nil
+	} else {
+		if oldStatus.LastPhaseTransitionTime != nil {
```

Comment:
This means that we will never allow changing the time once it's set regardless of feature gate.

I think just disallowing initial setting if the feature gate is off is sufficient.